### ISSUE TITLE
About the comparison between the anewell's and bearpaw's predictions on the validation set

### ISSUE THREAD

**Issue by zhiqiangdon** [anibali/eval-mpii-pose#1]
Hi,

Thanks for sharing your code. Do you run the https://github.com/anewell/pose-hg-demo/ to get he anewell's  prediction on the validation set? If so, I think the comparison is NOT totally comparable. Because the trick in file https://github.com/anewell/pose-hg-train/blob/master/src/util/pose.lua

`-- Very simple post-processing step to improve performance at tight PCK thresholds
    for i = 1,p:size(1) do
        for j = 1,p:size(2) do
            local hm = tmpOutput[i][j]
            local pX,pY = p[i][j][1], p[i][j][2]
            scores[i][j] = hm[pY][pX]
            if pX > 1 and pX < opt.outputRes and pY > 1 and pY < opt.outputRes then
               local diff = torch.Tensor({hm[pY][pX+1]-hm[pY][pX-1], hm[pY+1][pX]-hm[pY-1][pX]})
               p[i][j]:add(diff:sign():mul(.25))
            end
        end
    end
    p:add(0.5)`

is not used in the getPreds function of file https://github.com/anewell/pose-hg-demo/blob/master/util.lua
. Besides, the transform function in https://github.com/anewell/pose-hg-train/blob/master/src/util/img.lua is kind of different from transform function in https://github.com/anewell/pose-hg-demo/blob/master/img.lua. 

The bearpaw's prediction are based on the code in https://github.com/anewell/pose-hg-train rather than https://github.com/anewell/pose-hg-demo, which should have more advantages.

**Comment by anibali** [comment]
I simply followed the advice of the author in https://github.com/anewell/pose-hg-demo/issues/1:

> You can download the 8-stack model though and run the evaluation to see how it does.

Since pose-hg-demo is the official evaluation project released by the authors, I think that it makes sense to use that to generate the predictions. And yes, that is exactly what I did. I feel like it's as close to "official" validation set results as we'll get.

If you run a comparison between the two versions of anewell's evaluation with the official pretrained model, please do post the validation set file and results here in this thread. I'd be curious to see if it matters much. My feeling is that it won't have a big impact for PCKh thresholded at 0.5.

I'll also just make a note here that there are other subtle differences between bearpaw's and anewell's implementations, like different colour augmentation amounts during training.

**Comment by zhiqiangdon** [comment]
Thanks @anibali 

I will do this when I have enough time. It seems not that straight forward and I need to review some lua syntax. I think they should be some different. You see that the current PCKh is pretty high that even 1 percent gap makes a difference.
